From fa530556faac05d3e26c96eedc2fc1b899ca3e7f Mon Sep 17 00:00:00 2001
From: Mark Syms <mark.syms@citrix.com>
Date: Thu, 24 May 2018 09:47:31 +0100
Subject: [PATCH] CIFS: 511c54a2f69195b28afb9dd119f03787b1625bb4 adds a check
 for session expiry, status STATUS_NETWORK_SESSION_EXPIRED, however the server
 can also respond with STATUS_USER_SESSION_DELETED in cases where the session
 has been idle for some time and the server reaps the session to recover
 resources.

Handle this additional status in the same way as SESSION_EXPIRED.
---
 fs/cifs/smb2ops.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/fs/cifs/smb2ops.c b/fs/cifs/smb2ops.c
index 09254f3..0d63bb0 100644
--- a/fs/cifs/smb2ops.c
+++ b/fs/cifs/smb2ops.c
@@ -972,10 +972,11 @@ smb2_is_session_expired(char *buf)
 {
 	struct smb2_hdr *shdr = (struct smb2_hdr *)buf;
 
-	if (shdr->Status != STATUS_NETWORK_SESSION_EXPIRED)
+	if (shdr->Status != STATUS_NETWORK_SESSION_EXPIRED &&
+	    shdr->Status != STATUS_USER_SESSION_DELETED)
 		return false;
 
-	cifs_dbg(FYI, "Session expired\n");
+	cifs_dbg(FYI, "Session expired or deleted\n");
 	return true;
 }
 
-- 
1.8.3.1

